### PR TITLE
fix(components): [input-number] add Judging if the max is greater than the min

### LINKED ISSUE #12690: [Component] [input-number] el-input-number 组件max值比min小时，build后的代码，页面会直接崩溃（dev无异常）
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.47`
- Element Plus Version: `2.3.4`
- Browser / OS: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-input-number`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuY29uc3QgbnVtID0gcmVmKDApO1xuXG5jb25zdCBtc2cgPSByZWYoJ0hlbGxvIFdvcmxkIScpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8aDE+e3sgbXNnIH19PC9oMT5cbiAgPGVsLWlucHV0LW51bWJlciB2LW1vZGVsPVwibnVtXCIgOnByZWNpc2lvbj1cIjBcIiA6bWluPVwiMTAwXCIgOm1heD1cIjlcIiAvPlxuXG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce

```
pnpm create vite my-vue-app --template vue-ts
pnpm install
pnpm install element-plus
在main.ts添加element-plus
在vue文件中添加<el-input-number v-model="num" :min="100" :max="9" />

pnpm build
pnpm preview
打开预览，页面崩溃
```

### What is Expected?

页面不崩溃

### What is actually happening?

页面崩溃

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by zhengsixsix**
用户在传max的时候可能会小于min 会导致栈溢出 从而打包崩溃 增加判断防止这些问题

closed #12690

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by btea** [comment]
cc @element-plus/backers

**Comment by zhengsixsix** [comment]
> 用户在传max的时候可能会小于min 会导致栈溢出 从而打包崩溃 增加判断防止这些问题
> 
> 闭 #12690
> 
> 请确保在提交 PR 之前选中这些框，谢谢！
> 
> * [x] 确保您遵循贡献指南[英语](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) |（[中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) |[西班牙语](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) |[法兰西](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)）。
> * [x] 确保将提交合并到分支。`dev`
> * [x] 添加一些描述并参考 PR 的相关问题。

此描述不够准确 正确是  用户在传max的时候可能会小于min 执行build之后 再执行preview  页面是白屏的

File: packages/components/input-number/src/input-number.vue [hunk 1]
```diff
@@ -196,6 +196,11 @@ const verifyValue = (
   update?: boolean
 ): number | null | undefined => {
   const { max, min, step, precision, stepStrictly, valueOnClear } = props
+  if (max < min) {
+    throw new Error(
+      'InputNumber The maximum value of a component is smaller than the minimum values'
+    )
+  }
```
Comment (btea) [inline packages/components/input-number/src/input-number.vue:203]: Maybe you should use this function
https://github.com/element-plus/element-plus/blob/213fdfb51e51ec8a29bba4317e29a2b2ea25aa3f/packages/utils/error.ts#L10-L12

Comment (zhengsixsix) [reply packages/components/input-number/src/input-number.vue:203]: if (max < min) {
    debugWarn(
      'InputNumber',
      'The maximum value of a component is smaller than the minimum values'
    )
    return null
  }
Is this way of writing all right with you?

Comment (btea) [reply packages/components/input-number/src/input-number.vue:203]: IMO,  just replace it with `throwError` as written above. Maybe you can refer to here
https://github.com/element-plus/element-plus/blob/213fdfb51e51ec8a29bba4317e29a2b2ea25aa3f/packages/components/slider/src/composables/use-watch.ts#L33-L35

**Review by btea — APPROVED** [review]

**Review by ryuhangyeong — APPROVED** [review]

**Comment by ryuhangyeong** [comment]
image

test passes but a warning is displayed. can you solve this?

**Comment by zhengsixsix** [comment]
> image
> test passes but a warning is displayed. can you solve this?



> image
> test passes but a warning is displayed. can you solve this?

I seem unable to solve it. The warning message should be caused by the Vue.js framework. The error handler of the Vue.js framework will capture errors and print warning messages. I don't know if the information I found is accurate. In addition, the Slider component also has the same issue.

**Comment by btea** [comment]
@ryuhangyeong I think as @zhengsixsix  said, this is a deliberately thrown error, and maybe there is no need to solve it.

**Comment by ryuhangyeong** [comment]
@btea thanks for the explanation!

**Pull request merged**